src/fio/FioEngine.ts

```diff
@@ -384,212 +377,38 @@ export class FioEngine extends CurrencyEngine<FioTools, SafeFioWalletInfo> {
     actor: string,
     currencyCode: string = this.currencyInfo.currencyCode
   ): number {
-    const {
-      act: { name: trxName, data, account, authorization }
-    } = action.action_trace
-    let nativeAmount
-    let actorSender
-    let networkFee = '0'
-    let otherParams: TxOtherParams = {
-      account,
-      name: trxName,
-      authorization,
-      data,
-      meta: {}
-    }
-    const ourReceiveAddresses = []
-    if (action.block_num <= this.otherData.highestTxHeight) {
-      return action.block_num
+    const denom = getDenomination(
+      currencyCode,
+      this.currencyInfo,
+      this.allTokensMap
+    )
+    if (denom == null) {
+      this.error(`Received unsupported currencyCode: ${currencyCode}`)
+      return 0
```

Comment:
Unrelated to your changes and just something I noticed, but if FIO ever adds tokens this routine will break early because this number is a block height and it's compared against a local saved value.